docs/details/develop-checks.md

```diff
@@ -1,266 +1,313 @@
-- **Query** (string, optional) contains a SQL query or MongoDB query document (as a string with proper quoting) which is executed on the PMM Client side. It may be absent if type defines the whole query by itself.
-- **Script** (string, required) contains a small Python program that processes query results, and returns check results. It is executed on the PMM Server side.
-
-Check script assumes that there is a function with a fixed name _check_ that accepts a _list_ of _dicts_ containing returned rows for SQL databases and documents for MongoDB. It returns zero, one, or several check results that are then converted to alerts.
-
-Another function that should be implemented is **check_context**. PMM 2.12.0 and earlier use **context**, while newer versions use **check_context**. Both have the same meaning.
-
-The single query means that currently you cannot implement some advanced checks that would require several queries (and can't be implemented using SQL UNION).
-
-Checks format and the current STT UI use different terminology for severities. Here is how different formats show up on the UI:
-
-| Format    | UI       |
-| --------- | -------- |
-| emergency |          |
-| alert     |          |
-| critical  |          |
-| error     | Critical |
-| warning   | Major    |
-| notice    | Trivial  |
-| info      |          |
-| debug     |          |
-
+  
 ## Backend
 
-![!](../_images/BackendSTT.png)
-
-1. pmm-managed checks that this installation is opted-in into STT.
-2. pmm-managed downloads checks file from SaaS.
+1. pmm-managed checks that the installation is opted-in for checks.
+2. pmm-managed downloads files with checks from SaaS.
 3. pmm-managed verifies file signatures using a list of hard-coded public keys. At least one signature should be correct.
 4. pmm-managed sends queries to pmm-agent and gathers results.
 5. pmm-managed executes check scripts that produce alert information.
-6. pmm-managed sends alerts to Alertmanager.
-   - Due to Alertmanager design, pmm-managed has to send and re-send alerts to it much more often than the frequency of check execution. That's expected, and not important for using STT, but important for understanding how it works.
+6. pmm-managed sends alerts to Alermanager.
+   - Due to Alertmanager design, pmm-managed has to send and re-send alerts to it much more often than the frequency with which checks are executed. This expected behaviour is not important for using checks but is important for understanding how checks work.
    - Currently, Prometheus is not involved.
 
-## Frontend
-
-![!](../_images/FrontEndSTT.png)
+![!](../_images/BackendChecks.png)
 
-Our UI in Grafana uses Alertmanager API v2 to get information about failed security checks.
+## Frontend
+PMM UI in Grafana uses Alermanager API v2 to get information about failed checks:
 
+![!](../_images/FrontEndChecks.png)
+ 
+ 
+## Check fields
+Checks include the following fields:  
```

Comment:
We don't describe field "family". Also we don't have comparison V1 vs V2 formats.